### ISSUE TITLE
Allowing user-facing functions to be called programmatically

### ISSUE THREAD

**Issue by EFLS** [pprevos/citar-denote#33]
Hi Peter

I've been exploring your package recently and think it's really useful. One thing I'm missing currently, is a way to call the user-facing functions (like `citar-denote-find-citation`) to be called programmatically by passing a key to it. This would make integration with Embark easier.

I've modified the `citar-denote-find-citation` as follows (the only change is the optional argument and the `citekey` in the `if-let*` form):

```emacs-lisp
(defun citar-denote-find-citation (&optional key)
  "Find a Denote file that cites a bibliographic entry.

Provides a selection list of all bibliographic entries cited in Denote files."
  (interactive)
  (if-let* ((citations (citar-denote--extract-citations))
            (citekey (or key
                         (citar-select-ref
                           :filter (citar-denote--has-citekeys citations))))
            (files (citar-denote--retrieve-cite-files citekey)))
      (progn (find-file (denote-get-path-by-id
                         (denote-extract-id-from-string
                          (if (= (length files) 1)
                              (car files)
                            (denote-link--find-file-prompt files)))))
             (goto-char (point-min))
             (search-forward citekey))
    (message "Not citations found in Denote file")))
```

Is this something that makes sense? Or is there another way to achieve this?

The reason for doing so, is that now it is trivial to add something like:

```emacs-lisp
(define-key citar-embark-citation-map "c" 'citar-denote-find-citation)
```

to include an embark action that works on a key.

Cheers
EFLS

PS: Thanks for including a manual in Emacs and explaining so in the readme, that's really useful.

**Comment by pprevos** [comment]
Hi ELFS,

I don't use Embark as I find it confusing, but happy to enable Citar Denote to make it work.

Your version works great. Which other commands do you think need to be embark enabled?

Will you submit a pull request?

P:)

**Comment by EFLS** [comment]
Hi Peter

Thanks. It's not just for Embark though, it also helps users to build on your code, because now you can pass a citekey to functions like these.

I'll look into the codebase some more to see which other functions might make sense to change in this way. I might be able to submit a PR in a couple of days.

**Comment by EFLS** [comment]
Dag Peter

I submitted PR #36 specifically for this function, with some more tweaks. I think this approach, which makes the function prompt for user selection when called interactively, is better than what I suggested above.

Feel free to make changes as you see fit. One thing I haven't tested yet, is what happens when there are no citations in Denote files at all (in which case the `user-error` should be triggered).

Groeten
Elias
